### ISSUE TITLE
RO-Crates: credititing institutes and persons by name

### ISSUE THREAD

**Issue by cymon** [emo-bon/metaGOflow-data-products-RO-crate-example#4]
Christina mentioned that she wanted to ensure persons/institutes get credited by name in the ro-crates rather that just a blanked reference to the "EMO BON" project.

OK, this can be done, but who do we want to credit directly in the ro-crate?

Do we want to credit the collectors of the sample and the creators of the data? Or just the later and the leave the crediting of the sample collection to the sample event metadata records?

My suggestion is:

1) the person(s) at the station who took the sample, separate stanza for each person:

**"wasAssociatedWith"**:  {"@id": "#MatthiasObst"},

Is there a better label that wasAssociatedWith? Is there a "inCollaborationWith" or something similar?

        {
            "@id": "#MatthiasObst",
            "@type": "Person",
            "name": "Matthias Obst",
            "identifier": "https://orcid.org/0000-0003-0264-9631",
            "affiliation": {"@id": "https://ror.org/03f0f6041"}
        },

In the observatory Water_Column/Sediment Google logsheet, sampling sheet:
where _id/label_ is "sampl_person" and "other_person" (if recorded)
_name_ is "sampl_person" and "other_person" (if recorded) - two separate stanzas
**_email_ is UNKNOWN and not recorded.** I think this is a good thing
_identifier_ is "sampl_person_orcid" or 2other_person_orcid"
"affiliation" is the station the person is affiliated with

the station affiliation of the person(s) responsible for collecting the sample:
e.g. 
        {
            "@id": "https://ror.org/03f0f6041",
            "@type": "Organization",
            "name": "Flanders Marine Institute",
            **"url": "https://www.vliz.be/en",**
        },

In Observatory sheet in the Google logsheet:
where _id_ is the ROR id from the [API](https://ror.readme.io/docs/api-single)
where _name_ is "organization"
**There is no URL recoded for the stations in the Observatory sheets, so this cannot be included until that information is recorded. @cpavloud** or we just leave it out

2) The [data for what samples were analysed by which person/station are in this sheet:](https://docs.google.com/spreadsheets/d/1j9tRRsRCcyViDMTB1X7lx8POY1P5bV7UijxKKSebZAM/edit?gid=0#gid=0)

Basically the same as above but with:
the person at station who did the MetaGOflow analyses:
**"creator"**: {"@id": "#MatthiasObst"},
etc...
Currently that is only me and Stelios so we could hard-code that for the time-being.

4) the station responsible for performing the MetaGOflow analysis:
e.g.  the same as 1) above if different from station collecting sample

**Thoughts?**

**Comment by kmexter** [comment]
Yes, for 1 that will be dealt with as @laurianvm and I get round to ensuring good ro-crates for the observatory repos - on our list of todo for this year (but the list of todo for FE/EMO BON is longer than the time between now and xmas so some things will not get done)
For 2 - don't understand, is that not the same as 4?
For 4 -  so we have executor that was requested by ... someone ... and the term for that is being looked for by Laurian (see the issue on you at https://github.com/emo-bon/metaGOflow-data-products-RO-crate-example/issues/2). Is that enough? that can be a person and an institute

**Comment by cymon** [comment]
> Yes, for 1 that will be dealt with as @laurianvm and I get round to ensuring good ro-crates for the observatory repos - on our list of todo for this year (but the list of todo for FE/EMO BON is longer than the time between now and xmas so some things will not get done)

OK, not sure what means: but I'm going to code it up as shown and we can change/merge it later with what you do...

> For 2 - don't understand, is that not the same as 4?

Yeah, just left over formatting. It's the same format a 1) but instead of referencing the sample collection persons/station with "wasAssocatedWith", we're referencing the metaGOflow analysis "creator" and their station.

For 4 - so we have executor that was requested by ... someone ... and the term for that is being looked for by Laurian (see the issue on you at #2). Is that enough? that can be a person and an institute

Yeah, I think that's it (if I understand you correctly).

**Comment by cymon** [comment]
@kmexter @laurianvm

I've added "person" stanza's for the sampling_event_person (sampl_person in the logsheets) and the creator_person (who run the MGF analysis) and their respective "organization" stanza with Seadatanet EMOID links.

Also made the full links to the prototype S3 store at s3.mesocentre.uca.fr, and the links to the ENA accessions using the "downloadUrl" tag - basically anything that should have a link now has a link - even though the data may be missing.

K., can you take a look at the current 3 examples in [./ro-crates-examples](https://github.com/emo-bon/metaGOflow-data-products-RO-crate-example/tree/main/ro-crates-examples)?

**Comment by kmexter** [comment]
Looking at https://github.com/emo-bon/metaGOflow-data-products-RO-crate-example/blob/main/ro-crates-examples/EMOBON00224-ro-crate/ro-crate-metadata.json

- "MetaGoFlow Results for EMO BON Sampling Event EMOBON00224" -> better to use the sample ID here, not the geneoscope ID. So that would be "sampling event XXXX with Genoscope sequence ID EMOBON002244" (later to be replaced with the ENA sample accession number). ditto for the other places you use this ID. 
- the agents are nice but I am not sure you are doing it properly. See how we do it in https://github.com/arms-mbon/data_release_001/blob/main/ro-crate-metadata.json using "@id"= "_:KatrinaExter" and then just using "_:KatrinaExter" wherever I am mentioned. @laurianvm or @cedricdcc can you check and comment?
- I think you just need   "license": "https://creativecommons.org/licenses/by/4.0/", for license, nothing else - 97-102  I dont see as being necessary
- @cedricdcc or @laurianvm line 144, should that not be ./run.yaml? or is that just for the files that really are on that repository - and these files are not, they are on the S3
- @cymon are _all_ the files on the S3 (just checking)? So that would mean that the only file in the repo would actually be the ro-crate itself and the triples we make from them (tbd tho if those triples are small enough!)
- line 127 @laurianvm I am wondering if there is a relationship we can add? In that, this is the source data on which most of the other files were created from, while 152, 160, 176 and others their relationship is that they are the outputs of the code referenced in line 135 
- 'scuse me if I missed it, but is the metagoflow  code itself referenced as being the code that was used (as opposed to the stanza 135 which I think is just the version?) 
- also, we need to link to the observatory repo that contains the sample metadata - "rekatedTo" @laurianvm ?
- and specifically flag the sample ID and accession number as specific objects - we have to think of how that triple statement would look, @laurianvm 

dashing off for a long weekend in France now....
-

**Comment by cymon** [comment]
On Thu, 31 Oct 2024 at 13:28, Katrina Exter ***@***.***>
wrote:

> Looking at
> https://github.com/emo-bon/metaGOflow-data-products-RO-crate-example/blob/main/ro-crates-examples/EMOBON00224-ro-crate/ro-crate-metadata.json
>
>    - "MetaGoFlow Results for EMO BON Sampling Event EMOBON00224" ->
>    better to use the sample ID here, not the geneoscope ID. So that would be
>    "sampling event XXXX with Genoscope sequence ID EMOBON002244" (later to be
>    replaced with the ENA sample accession number). ditto for the other places
>    you use this ID.
>
>
By "sample ID" I'm assuming you mean the "source_mat_id"s in the sampling
logsheets, right? This will now work as we have a sane (ie unique) labeling
scheme :)

So I'll rename the ro-crates to "<source_mat_id>-ro-crate".

So crates will now look like this:
e.g.

EMOBON_RFormosa_Wa_21-ro-crate
EMOBON_MBAL4_Wa_1-ro-crate
etc...

I'm averse to calling the ref_code the "Genoscope ID" as it wasn't assigned
by Genoscope - it was assigned by EMO BON (Ioulia) before sending to
Genoscope. So I'd rather just call it "the EMO BON reference code for the
molecular data".


>
>    - the agents are nice but I am not sure you are doing it properly. See
>    how we do it in
>    https://github.com/arms-mbon/data_release_001/blob/main/ro-crate-metadata.json
>    using ***@***.*** <https://github.com/id>"= "*:KatrinaExter" and then just
>    using "*:KatrinaExter" wherever I am mentioned. @laurianvm
>    <https://github.com/laurianvm> or @cedricdcc
>    <https://github.com/cedricdcc> can you check and comment?
>
> Yeah, as far as I'm aware any character(s) can be used to prefix and ID;
I've seen "#", ":" "_:" etc... it really isn't necessary as far as I can
tell as long as the ID stanza it's pointing to has the same value. You
could use a UUID for instance :)


>    - I think you just need "license": "
>    https://creativecommons.org/licenses/by/4.0/", for license, nothing
>    else - 97-102 I dont see as being necessary
>    - @cedricdcc <https://github.com/cedricdcc> or @laurianvm
>    <https://github.com/laurianvm> line 144, should that not be
>    ./run.yaml? or is that just for the files that really are on that
>    repository - and these files are not, they are on the S3
>
> I'm pushing all the metaGOflow data files to the S3 repository using DVC.

>
>    - @cymon <https://github.com/cymon> are *all* the files on the S3
>    (just checking)?
>
> Yes... well they will be once I've uploaded them...


>
>    - So that would mean that the only file in the repo would actually be
>    the ro-crate itself and the triples we make from them (tbd tho if those
>    triples are small enough!)
>
> This is my current plan for the ro-crate payload:
EMOBON_RFormosa_Wa_21-ro-crate/
                          ro-crate-metadata.json
                          ro-crate-preview.html
                          run.yml.dvc
                          fastp.html.dvc
                          etc for each and all MGF .dvc files in the same
structure as in the S3 repo

The DVC files are very small, but it means users can use DVC to grab the
data files as well as the download URLs in the ro-crate metadata.

This also means I can point to the DVC files in the MGF data stanzas and
the have the S3 URL's as downloadURLs:
        {
            ***@***.***": "./functional-annotation/DBH.merged.summary.go_slim.dvc",
            ***@***.***": "File",
            "name": "Merged contigs InterProScan slim",
            "downloadUrl": "
http://s3.mesocentre.uca.fr/mgf-data-products/HCFCYDSX5.UDI129%2Ffunctional-annotation%2FDBH.merged.summary.go_slim
",
            "encodingFormat": "text/plain"
        },

>
>    - line 127 @laurianvm <https://github.com/laurianvm> I am wondering if
>    there is a relationship we can add? In that, this is the source data on
>    which most of the other files were created from, while 152, 160, 176 and
>    others their relationship is that they are the outputs of the code
>    referenced in line 135
>    - 'scuse me if I missed it, but is the metagoflow code itself
>    referenced as being the code that was used (as opposed to the stanza 135
>    which I think is just the version?)
>
> Not sure what you mean: the version referenced is the version of the code
used - not sure what else would be necessary.

We have something along the lines of "description": "RO-Crate for the data
products from the MetaGOflow analysis of EMO BON Sampling Event
EMOBON00224","

>
>    - also, we need to link to the observatory repo that contains the
>    sample metadata - "rekatedTo" @laurianvm <https://github.com/laurianvm>
>    ?
>    - and specifically flag the sample ID and accession number as specific
>    objects - we have to think of how that triple statement would look,
>    @laurianvm <https://github.com/laurianvm>
>
> dashing off for a long weekend in France now....

Lucky you! Portuguese national holiday here tomorrow as well... Message ID:
<emo-bon/metaGOflow-data-products-RO-crate-example/issues/4/2449844373@
github.com

-- 
___________________________________________

Cymon J. Cox


Senior Researcher
Plant Systematics and Bioinformatics Digital Laboratory
Centro de Ciencias do Mar (CCMAR) - CIMAR-Lab. Assoc.

Mailing address:
CCMAR - Centro de Ciencias do Mar,
Universidade do Algarve
Campus de Gambelas
Edif. 7
8005-139 Faro
Portugal

Phone: +351 289800051 ext 7380
Fax: +351 289800051
Email: ***@***.***

@CCMAR <https://ccmar.ualg.pt/users/cymon> | Google Scholar
<https://scholar.google.co.uk/citations?user=f5M7DhkAAAAJ&hl=en&oi=ao> |
Scopus
<http://www.scopus.com/inward/authorDetails.url?authorID=7402112716&partnerID=MN8TOARS>
|
Orcid <http://orcid.org/0000-0002-4927-979X>  | CienciaVitae
<https://www.cienciavitae.pt/6B15-9771-1D04>
GPG: Public key on keyserver.ubuntu.com

___________________________________________

**Comment by cymon** [comment]
On Thu, 31 Oct 2024 at 14:42, Cymon J. Cox ***@***.***> wrote:

> On Thu, 31 Oct 2024 at 13:28, Katrina Exter ***@***.***>
> wrote:
>


> I think you just need "license": "
> https://creativecommons.org/licenses/by/4.0/", for license, nothing else
> - 97-102 I dont see as being necessary
>
>>
>>    - @cedricdcc <https://github.com/cedricdcc> or @laurianvm
>>    <https://github.com/laurianvm> line 144, should that not be
>>    ./run.yaml? or is that just for the files that really are on that
>>    repository - and these files are not, they are on the S3
>>
>> I'm pushing all the metaGOflow data files to the S3 repository using DVC.
>

Let me just clarify that: I'm pushing all the MGF data files that are part
of these data products ro-crates to S3. The remaining intermediate (and
large) sequence data files we have yet to decide what to do with.

-- 
___________________________________________

Cymon J. Cox


Senior Researcher
Plant Systematics and Bioinformatics Digital Laboratory
Centro de Ciencias do Mar (CCMAR) - CIMAR-Lab. Assoc.

Mailing address:
CCMAR - Centro de Ciencias do Mar,
Universidade do Algarve
Campus de Gambelas
Edif. 7
8005-139 Faro
Portugal

Phone: +351 289800051 ext 7380
Fax: +351 289800051
Email: ***@***.***

@CCMAR <https://ccmar.ualg.pt/users/cymon> | Google Scholar
<https://scholar.google.co.uk/citations?user=f5M7DhkAAAAJ&hl=en&oi=ao> |
Scopus
<http://www.scopus.com/inward/authorDetails.url?authorID=7402112716&partnerID=MN8TOARS>
|
Orcid <http://orcid.org/0000-0002-4927-979X>  | CienciaVitae
<https://www.cienciavitae.pt/6B15-9771-1D04>
GPG: Public key on keyserver.ubuntu.com

___________________________________________

**Comment by kmexter** [comment]
> By "sample ID" I'm assuming you mean the "source_mat_id"s in the sampling logsheets, right? This will now work as we have a sane (ie unique) labeling scheme :) So I'll rename the ro-crates to "<source_mat_id>-ro-crate". So crates will now look like this: e.g. EMOBON_RFormosa_Wa_21-ro-crate EMOBON_MBAL4_Wa_1-ro-crate etc... 

-->Grand!

>I'm averse to calling the ref_code the "Genoscope ID" as it wasn't assigned by Genoscope - it was assigned by EMO BON (Ioulia) before sending to Genoscope. So I'd rather just call it "the EMO BON reference code for the molecular data".

-->OK


> - line 127 @laurianvm <https://github.com/laurianvm> I am wondering if there is a relationship we can add? In that, this is the source data on which most of the other files were created from, while 152, 160, 176 and others their relationship is that they are the outputs of the code referenced in line 135 - 'scuse me if I missed it, but is the metagoflow code itself referenced as being the code that was used (as opposed to the stanza 135 which I think is just the version?) 
Not sure what you mean: the version referenced is the version of the code used - not sure what else would be necessary. 

--> if you have a field "version" that it should just have a number in it (or date), while the code itself is linked to another field - I think the stanza that is on line 262 of https://github.com/arms-mbon/analysis_release_001/blob/main/ro-crate-metadata.json can be used to point to both the software URL and its name, type, and version

For DVC stuff - I am sure you know better than I, I would be happy if @cedricdcc and @laurianvm could look at those stanzas and comment.

**Comment by cymon** [comment]
There are a bunch more examples of ro-crates in ./emo-bon-ro-crate-repository these are all the sediment analyses for batch 1 and 2.

I created a "ghost" archive for each run where all the files are present but the file contents are missing (except for the word "ghost") - this means all the links in the ro-crates metadata.json are functional. Of course, once we have a public S3 repo (and not just the loner from d4science that we are currently using), we can upload the real data - but this just prototype procedures effectively.

DVC - I coded up the functions to add dvc stubs archives: but to my mind there are several issues with using DVC

- the data that results from a MGF doesnt change (does not get updated) - so the main point of DVC to keep track of changing data sets is irrelevant
- adding DVC stubs to files and uploading to S3 means there are no typical download URLs to use a links in the metadata.json stanzas
- to down load the data from S3 using the dvc file, you need to install dvc and pull. So this is doable but will turn people off from using the data if they have to go through a dvc configuration. We could provide the dvc config in each ro-crate but again more flaff...
- the original idea was only to have "large" files in S3 and the "smaller" files on github, but dvc can only be initialised with git if the the dvc root folder is also the git root folder. This isnt going to work. We could make a separate repository for emo-bon-ro-crates and init both git and dvc in the root directory, but at this point it's getting really messy and difficult to explain to someone who just want to get the data

Basically, the alternative - ie not using dvc - just make life so much easier:
- we upload **all** of the data files (big and small) to the S3 store using s5cmd and use the public URLs as the download URLs in the stanza. This is what I've currently implemented, because it is just so much simpler.

I've still to swap out the names of the ro-crates for the sampl_mat_id's because that first requires me to re-run all the validation on the log sheets and create the combined tables which have changed considerably since I last did this... I'll get around to this when I can...

**Comment by kmexter** [comment]
@cymon - good point but a totally different issue. I am going to copy it out of this issue into a new one. @laurianvm and I will be looking at all emo bon ro-crates (the observatory ones in particular, as we have not done anything to them for a year) over the next few weeks (i.e. deadline xmas)

**Comment by cymon** [comment]
> I've still to swap out the names of the ro-crates for the sampl_mat_id's because that first requires me to re-run all the validation on the log sheets and create the combined tables which have changed considerably since I last did this... I'll get around to this when I can...

Done.

Check out new ro-crates in ./emo-bon-ro-crate-repository

**Comment by kmexter** [comment]
I think we can close this issue and carry on in https://github.com/emo-bon/metaGOflow-data-products-RO-crate-example/issues/7

**Comment by cymon** [comment]
OK, before I rip out the code that does this: are we sure?

I agree that it's not appropriate for it to be in the MGF data products ro-crate and is better left in the sampling metadata ro-crate only. But it's there now (in the MGF rc), and it's what Christina (and by implication EMBRC HQ) wanted.

Christina - can I pass you hill to make your final stand on?

**Comment by kmexter** [comment]
@cpavloud see comment from Cymon above
We are asking him to take out of the MGF ro-crates the mention of people who did the sampling, since those people are anyway in the observatory ro-crates and since a ro-crate should describe the current dataset, not one that it was derived from
This is not the same as acknowledgements made in metadata records that we will have harvested by BC and FE - those will be different files which we will put together based on both the mfg and the related observatory ro-crates

One should not be putting all possible things in a ro-crate as it then loses its purpose, which is to describe, to machines, what a data package consists of

**Comment by cpavloud** [comment]
In the observatory logsheets (and hence in the observatory RO-crates?) there is the name of the responsible person for the observatory. 
In the sampling logsheets (and hence in the sampling RO-crates?), there are the names of the people who did the sampling. So these should be there.
In the MGF RO-crate, which is our data products RO-crate, we should have the mention of the people who did the analysis (e.g. Cymon, Stelios), as well as the mention of the HPC where said analysis took place.

**Comment by cpavloud** [comment]
Probably we need the sampl_person only. Maybe it is too much to add all store_person and other_person

**Comment by cymon** [comment]
> For MGF, @cymon can do that one. I think we agreed that people who ran MFG are the creators (giving their full name and institute details) and EMO BON is as well.

You want 2 creators, "person who runs mgf" and EMO BON. Currently, only "person who runs mgf" is included in a "@type Person stanza" that links to the person's Orcid account. What do I link EMO BON to? The "@type": "Organization" EMBRC stanza? 

> The place where the analysis took place

Really, are you sure? I currently run MGF in 4 different places - do you want a reference to organisation that provided access to the computational infrastructure, or the organisation that maintains the facility, or both?

> - certainly if not the place that is the organisation of one of the creators - can be a wasAssociatedWith
That's how it was before I removed the code according to the instructions of Marc.

Can you take a look at a current ro-crate-metadata.json and provide exact code for the changes you want made - that would help a lot...

**Comment by kmexter** [comment]
To the first - yes, link emobon to embrc. But you can wait until @laurianvm is finished wtih the BPNS example - you can follow that. Next year is fine, Laurian (noting that we are all taking off the 2 weeks of xmas)
To the second - well, that was a request from @cpavloud, so discuss it with her
To the third - again, wait until Laurian is done with the BPNS one and we can provide that as an example.

**Comment by cpavloud** [comment]
>> The place where the analysis took place

> Really, are you sure? I currently run MGF in 4 different places - do you want a reference to organisation that provided access to the computational infrastructure, or the organisation that maintains the facility, or both?

All I am saying is that since we are using HPC resources, no matter where from, we need to make sure that we mention which were those resources. 
It is not enough that we know among ourselves. 

Which are the 4 different places? Isn't it just HCMR and CCMAR at this point? 

I would prefer a reference to the HPC facility itself. 
E.g. hpc_name: Zorbas 
hcp_citation: https://doi.org/10.1093/gigascience/giab053
hpc_website: https://hpc.hcmr.gr/

**Comment by cymon** [comment]
> > > The place where the analysis took place
> 
> > Really, are you sure? I currently run MGF in 4 different places - do you want a reference to organisation that provided access to the computational infrastructure, or the organisation that maintains the facility, or both?
> 
> All I am saying is that since we are using HPC resources, no matter where from, we need to make sure that we mention which were those resources. It is not enough that we know among ourselves.

OK, I'm not sure I agree; I don't see what useful information it provides.

> 
> Which are the 4 different places? Isn't it just HCMR and CCMAR at this point?

I run analyses at CCMAR on our own cluster CETA, my own research group server REDI, using an national computing facility INCD facility CIRRUS in Lisbon, and an INCD facility UID in Braga (northern Portugal).
As an aside, we should be looking to get an EGI project to analyse Batch 3 - we currently cannot analyse samples fast enough.

> I would prefer a reference to the HPC facility itself. E.g. hpc_name: Zorbas hcp_citation: https://doi.org/10.1093/gigascience/giab053 hpc_website: https://hpc.hcmr.gr/

OK, this can be done if these metadata are stored somewhere ([maybe in the same spreadsheet in a separate sheet](https://docs.google.com/spreadsheets/d/1j9tRRsRCcyViDMTB1X7lx8POY1P5bV7UijxKKSebZAM/edit?gid=0#gid=0)?) and the facility where the sample is analysed is recorded (which it is).

However, can you run this by Marc who is the "Keeper of RO-Crate Purity" to make sure he agrees that this information belongs in this ro-crate (and not somewhere else). Else I end up writing code that only later gets removed.
Also, some examples of the JSON stanzas you want to see would be very helpful.

**Comment by cpavloud** [comment]
@marc-portier can you check Cymon's comment above?
